compiler/rustc_lexer/src/lib.rs

```diff
@@ -151,24 +151,27 @@ pub enum LiteralKind {
     /// "b"abc"", "b"abc"
     ByteStr { terminated: bool },
     /// "r"abc"", "r#"abc"#", "r####"ab"###"c"####", "r#"a"
-    RawStr { n_hashes: u16, err: Option<RawStrError> },
+    RawStr { n_hashes: u16, valid: bool },
     /// "br"abc"", "br#"abc"#", "br####"ab"###"c"####", "br#"a"
-    RawByteStr { n_hashes: u16, err: Option<RawStrError> },
+    RawByteStr { n_hashes: u16, valid: bool },
 }
 
 /// Error produced validating a raw string. Represents cases like:
 /// - `r##~"abcde"##`: `InvalidStarter`
 /// - `r###"abcde"##`: `NoTerminator { expected: 3, found: 2, possible_terminator_offset: Some(11)`
 /// - Too many `#`s (>65535): `TooManyDelimiters`
+///
+/// This enum is 20 bytes big, bigger than the rest of `Token`. So we don't store it
+/// in `LiteralKind::RawStr`, instead you can get it with `recompute_raw_string_error()`.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
 pub enum RawStrError {
     /// Non `#` characters exist between `r` and `"` eg. `r#~"..`
     InvalidStarter { bad_char: char },
     /// The string was never terminated. `possible_terminator_offset` is the number of characters after `r` or `br` where they
     /// may have intended to terminate it.
-    NoTerminator { expected: usize, found: usize, possible_terminator_offset: Option<usize> },
+    NoTerminator { expected: u32, found: u32, possible_terminator_offset: Option<u32> },
```

Comment:
Could this be `u16`? I think it is hard for raw string to hit `u32` in source code which editor would have a hard time handling it? Or would it? I think this is a less safe choice compared to the reductions I mentioned above but looks possible.

Reply:
It doesn't matter here because `RawStrError`s aren't stored anymore and aren't constructed that often. I also didn't want to regress potential errors in weird cases.